modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts

```diff
@@ -55,9 +55,21 @@ const nestedBlockMerge = (
 
 const sidelongBlockMerge = (rootNode: SugarElement<Node>, fromBlock: SugarElement<Element>, toBlock: SugarElement<Element>): Optional<CaretPosition> => {
   if (Empty.isEmpty(toBlock)) {
+    const inlineDescendants = Traverse.firstChild(toBlock).fold(
+      Fun.constant([]),
+      (child) => {
+        const descendants = PredicateFilter.descendants(child, ElementType.isInline);
+        return ElementType.isInline(child) ? [ child, ...descendants ] : descendants;
+      }
+    );
+
     Remove.remove(toBlock);
     if (Empty.isEmpty(fromBlock)) {
-      PaddingBr.fillWithPaddingBr(fromBlock);
+      const bogus = PaddingBr.fillWithPaddingBr(fromBlock);
+      Arr.foldr(inlineDescendants, (descendant: SugarElement<HTMLElement>, wrapper) => {
+        Insert.wrap(descendant, wrapper);
```

Comment:
This still doesn't look right. You said `The idea is to wrap the bogus with toBlock descendants`, but now you're wrapping descendants with the `br` tag.

My main point with the code yesterday was that your result remains in `toBlock`; I would expect something like
* move descendant into `fromBlock`
* move the BR into the descendant
* something in this process that ensures the order of the nested structure is retained

Reply:
I'm utterly confused. If this code worked as you're describing there will be nothing in the editor. As well as the tests won't pass. Everything works as expected though. Also how cna one wrap something with a `br` tag? `br` is always the deepest descendant in the `Insert.wrap` call.

I've updated the code with an alternative solution, explicitly cloning children from the `toBlock` with `Replication` API and inserting the bogus to the deepest child. Hoping this updated version will clarify things a bit.

https://github.com/tinymce/tinymce/blob/1c83d5b1e2668bb0b56985b949bafc0bbad39fc1/modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts#L58-L72

Reply:
A couple of things happened here. I was wrong, your choice of variable names and a cursory glance at the modules it references lead me to an incorrect conclusion. But after further investigation _it's even more confusing_.

Please revert the latest commit and let's try again. You were actually quite close to what I expected.

What happened is I misunderstood the `bogus` const. We usually reserve that variable name for references to the actual bogus BR tag, and that's what it appeared to be looking at the PR, but that's not what you did. Perhaps you thought it was `fromBlock`, which would make your correct. But it was undefined - `fillWithPaddingBr` _does not return a value_.
https://github.com/tinymce/tinymce/blob/ece3bf774b9ae20910d073af5351f162c4fb52d5/modules/tinymce/src/core/main/ts/dom/PaddingBr.ts#L32-L35

I'm not sure how your code did anything, because it should've resulted in references to `bogus.dom` which is a runtime error when `bogus` is `undefined`.

So, here's what I think we should do, rather than your changes to meet my incorrect advice:
* Remove the `bogus` variable
* pass `fromBlock` to `Arr.foldr` as the wrapper
* write more tests, because I suspect the similarity of the two empty paragraphs is why it passes incorrectly. And make sure to do the red/green cycle - disable your fix and see if the tests fail in the way you expect.

That should be better.

Reply:
Wait, wait, you modified `fillWithPaddingBr` and reversed it in that commit. So I'm not going crazy. But your variable names were wrong - due to how `Arr.foldr` works `descendant` is the bogus element, `wrapper` is the descendant. And so we return to your code being confusing.

I'm not sure what to recommend now. Still revert the commit, but try renaming the variables and instead of modifying `fillPaddingBr` just put that code here. Yes your changes didn't break existing usages of the function, but I as someone who expects the function to do a certain thing was confused about how your code operated and _that_ is the problem. Code should not be surprising.

Reply:
> But `bogus` is the **descendant** and `toBlock`'s descendant is the **wrapper** of the `bogus` in this context. 

That makes sense _inside_ the function. But looking at the full context, when you're iterating over an array called `inlineDescendants`, we normally use words like `descendant` in the function argument to refer to elements of the array. Suddenly reversing that, while technically correct, is logically confusing.

One way I deal with this (naming is hard) is by using the generic `acc` name for the accumulator, as the implementation of `foldr` does. Then I don't need to describe what the accumulator is holding, only that it's the accumulator.